qiskit_nature/operators/second_quantization/vibrational_spin_op_utils/vibrational_labels_validator.py

```diff
@@ -109,12 +109,12 @@ def _is_order_incorrect(mode_index: int, prev_mode_index: int, modal_index: int,
 
 
 def _is_mode_order_incorrect(mode_index: int, prev_mode_index: int) -> bool:
-    return mode_index > prev_mode_index
+    return mode_index < prev_mode_index
```

Comment:
Maybe I am missing some context here but why does the current `mode_index` need be smaller than the previous one?

Reply:
Ah this method is called ...*in*correct..
Is there a particular reason for the code being this way around? If yes, please add a single sentence docstring or comment.

Reply:
I am not sure the `VibrationalSpinOp`. @dlasecki What do you think this?

Reply:
So the way @ikkoham parses these in SpinOp required a descending order in terms of modes and modals. When changing endianness, this needs to be reversed.

Reply:
I rather meant that I find it a little bit awkward to have a function which check whether something is _incorrect_ rather than whether something is _correct_. But this is probably just personal preference.
You can disregard this comment for the purposes of this PR. The code fix is definitely correct :+1: